widget_driver/README.md

```diff
@@ -372,6 +411,67 @@ Easy...
     }
    ```
 
+### Caveat's when working with dependencies
+
+The driver is bound to the lifecycle of the widget, this means it lifes as long as the widget does. However we do not want to recreate the driver on every state change, that would increase the build time of a `DrivableWidget`. (That's also why we need the mixin for `@driverProvidableProperties` annotated properties)
+However we want to resolve our dependencies in our constructor. (Using e.g. `Provider`) So how do those get updated?
+We tied the recreation of the driver to the `didChangeDependencies` statefunction. Should you watch, listen or subscribe to updates to your dependencies, we will rebuild the driver for you. Thus allowing you to re-resolve your service and create new listeners etc.
+As Google's `Provider` package is one of the most used packages in that category, here is an example:
+
+
+#### Example with Google Provider
+
+Say we have a Auth-Service, which could be changed at runtime.
+If we were to resolve it like this:
+
+```dart
+class SomeDriver extends WidgetDriver {
+  final AuthService _service;
```

Comment:
I am not sure if a service is the best example for showing this issue..
It might confuse people to think that the data changes coming from the service might be lost..

But the issue relates to the actual service instance changing.. 
And as you mentioned, services are more likely not changing..
A model data example would maybe fit better here..

Reply:
for example take ThemeData for checking light or darkmode

Reply:
I called it a ThemeDataService which does a bit more than just ThemeData. Reason being, that ThemeData is already resolved from the BuildContext, I however don't know if it has a notifing "characteristic".